src/test/java/bio/overture/ego/controller/UserControllerTest.java

```diff
@@ -241,12 +251,287 @@ public void createUser_EmailAlreadyExists_Conflict() {
+            .type(randomEnum(UserType.class))
+            .preferredLanguage(randomEnum(LanguageType.class))
+            .firstName(randomStringNoSpaces(10))
+            .lastName(randomStringNoSpaces(10))
+            .identityProvider(GITHUB)
+            .providerId(nonExistentProviderId)
+            .build();
+
+    // Create the user and assert OK
+    createUserPostRequestAnd(r).assertOk();
+  }
+
+  @Test
+  public void createUser_ExistingProviderNonExistingProvId_OK() {
+    // Generate data
+    val data = generateUniqueTestUserData();
+    val user0 = data.getUsers().get(0);
+
+    val nonExistingProvId = generateNonExistentProviderId(userService);
+    // Create a request with an existing provider and nonexisting providerId
+    val r =
+        CreateUserRequest.builder()
+            .email("user0@email.com")
+            .status(randomEnum(StatusType.class))
+            .type(randomEnum(UserType.class))
+            .preferredLanguage(randomEnum(LanguageType.class))
+            .firstName(randomStringNoSpaces(10))
+            .lastName(randomStringNoSpaces(10))
+            .identityProvider(user0.getIdentityProvider())
+            .providerId(nonExistingProvId)
+            .build();
+
+    createUserPostRequestAnd(r).assertOk();
+  }
+
+  @Test
+  public void createUser_ExistingProviderIdNonExistingProvider_OK() {
+    // Generate data
+    val data = generateUniqueTestUserData();
+    val user0 = data.getUsers().get(0);
+
+    // Create a request with an existing provider and nonexisting providerId
+    val r =
+        CreateUserRequest.builder()
+            .email("user0@email.com")
+            .status(randomEnum(StatusType.class))
+            .type(randomEnum(UserType.class))
+            .preferredLanguage(randomEnum(LanguageType.class))
+            .firstName(randomStringNoSpaces(10))
+            .lastName(randomStringNoSpaces(10))
+            .identityProvider(LINKEDIN)
+            .providerId(user0.getProviderId())
+            .build();
+
+    createUserPostRequestAnd(r).assertOk();
+  }
+
+  @Test
+  public void createUser_ExistingProviderIdExistingProvider_Conflict() {
+    // Generate data
+    val data = generateUniqueTestUserData();
+    val user0 = data.getUsers().get(0);
+
+    // Create a request with a non-unique identityProvider + providerId
+    val r =
+        CreateUserRequest.builder()
+            .email("user0@email.com")
+            .status(randomEnum(StatusType.class))
+            .type(randomEnum(UserType.class))
+            .preferredLanguage(randomEnum(LanguageType.class))
+            .firstName(randomStringNoSpaces(10))
+            .lastName(randomStringNoSpaces(10))
+            .identityProvider(user0.getIdentityProvider())
+            .providerId(user0.getProviderId())
             .build();
 
-    // Create the user and assert a conflict
     createUserPostRequestAnd(r).assertConflict();
   }
 
+  @Test
+  public void createUser_InvalidProvider_BadRequest() {
```

Comment:
Also add similar tests for UNKNOWN providerTypes. If the logic, in the comments before, is implemented, we want to test that execution path, where users prior to the migration, can have their providerType updated.